warp/src/main/java/com/schibsted/nmp/warp/components/WarpStepIndicator.kt

```diff
@@ -1,281 +1,580 @@
+        maxWidth = constraints.maxWidth / steps
+    )
+    val labelPlaceables = measurables.filter { it.layoutId == StepIndicatorIds.LABEL }.map { it.measure(labelConstraint) }
+    val indicatorPlaceables = measurables.filter { it.layoutId == StepIndicatorIds.INDICATOR}.map {
+        it.measure(
+            Constraints(
+                minWidth = dimensions.placeholderIndicatorSize.dp.roundToPx(),
+                maxWidth = dimensions.placeholderIndicatorSize.dp.roundToPx(),
+                minHeight = dimensions.placeholderIndicatorSize.dp.roundToPx(),
+                maxHeight = dimensions.placeholderIndicatorSize.dp.roundToPx()
+            )
+        )
+    }
+    val indicatorWidth = indicatorPlaceables.first().width
+
+    // Check if the labels or indicators on the end are widest
+    val maxLabel = max(labelPlaceables.firstOrNull()?.width ?: 0, labelPlaceables.lastOrNull()?.width ?: 0) / 2 - indicatorWidth / 2
+    val endSpace = max(indicatorWidth / 2, maxLabel)
+
+    val space = (width - indicatorWidth * steps - 2 * endSpace) / (steps - 1)
+    val linePlaceables = measurables.filter { it.layoutId == StepIndicatorIds.TRACK}.map {
+        // Adjust the width just a little so we do not get gaps from the curvature of the indicators
+        it.measure(constraints.copy(minWidth = space + 2, maxWidth = space + 2))
+    }
+
+    val labelHeight = labelPlaceables.maxByOrNull { it.height }?.height ?: 0
+
+    val height = indicatorPlaceables.first().height + labelHeight
+
+    layout(width, height) {
 
+        linePlaceables.forEachIndexed { index, placeable ->
+            val xPos = endSpace + space * index + indicatorWidth * (index + 1) - 1
+            placeable.place(xPos, labelHeight + indicatorWidth / 2)
+        }
+        indicatorPlaceables.forEachIndexed { index, placeable ->
+            val xPos = endSpace + space * index + indicatorWidth * index
+            placeable.place(xPos, labelHeight)
+        }
+        labelPlaceables.forEachIndexed { index, placeable ->
+            val xPos = space * index + indicatorWidth * index
+            placeable.place(xPos, 0)
+        }
 
-enum class StepState {
-    Active,
-    Completed,
-    Unvisited,
-    // TODO peeked?
+    }
 }
 
+/**
+ * A component for showing progress through a series of steps.
+ * @param modifier Modifier to be applied to the component
+ * @param steps Number of steps to display in the component. Must be greater than 1
+ * @param activeStep The currently active step
+ * @param onStepClicked Optional callback for when a step is clicked
+ * @param label Optional composable to layout above each step. The argument to the function is the index of the step
+ */
 @Composable
-fun Dot(
+fun HorizontalWarpStepIndicator(
     modifier: Modifier = Modifier,
-    state: StepState,
-    onClick: (() -> Unit)?
+    steps: Int,
+    activeStep: Int = 0,
+    onStepClicked: ((Int) -> Unit)? = null,
+    label: @Composable ((Int) -> Unit)? = null
```

Comment:
I think this should just be a string so the the styling of the label is in our control